### ISSUE TITLE
ZeroDivisionError: integer division or modulo by zero

### ISSUE THREAD

**Issue by ederenn** [golemfactory/clay#3366]
depends on: #3370 

#### Encountered on
Mac OS, source, GOLEM Version: 0.17.1+dev330.gfcfdf5b

during starting golem

```
ERROR    [golem.node                         ] Stopping because of 'setup client' error: <twisted.python.failure.Failure builtins.ZeroDivisionError: integer division or modulo by zero>
Traceback (most recent call last):
  File "/Users/ederenn/golem/venv/lib/python3.6/site-packages/twisted/internet/defer.py", line 653, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/Users/ederenn/projects/golem/golem/node.py", line 372, in _setup_client
    self.client = self._client_factory(self._keys_auth)
  File "/Users/ederenn/projects/golem/golem/node.py", line 111, in <lambda>
    task_finished_cb=self._try_shutdown
  File "/Users/ederenn/projects/golem/golem/client.py", line 205, in __init__
    Path(self.datadir))
  File "/Users/ederenn/projects/golem/golem/ethereum/fundslocker.py", line 34, in __init__
    self.restore()
  File "/Users/ederenn/projects/golem/golem/ethereum/fundslocker.py", line 85, in restore
    task.num_tasks,
  File "/Users/ederenn/projects/golem/golem/ethereum/transactionsystem.py", line 386, in lock_funds_for_payments
    (self._payments_locked + self._payment_processor.recipients_count)
ZeroDivisionError: integer division or modulo by zero

**Comment by ederenn** [comment]
logs from dubug mode
including task logs previous to error
https://drive.google.com/open?id=1btRdW90rhZsAfmOSg0cQmvDyHnc8Oe91

[golem-7.log](https://github.com/golemfactory/golem/files/2433834/golem-7.log)

**Comment by Krigpl** [comment]
Root cause: #3370
In other words this happens when all subtasks are marked as finished but the task itself is not.

**Comment by mplebanski** [comment]
This happens to me also on task creation:

[golem.log](https://github.com/golemfactory/golem/files/2449555/golem.log)

This is just some PoC task but it's not relevant in this case. My node is brand new (fresh datadir).

**Comment by Krigpl** [comment]
@mplebanski 
> My node is brand new (fresh datadir).

Your logs suggest this is not a fresh datadir at all.

**Comment by mplebanski** [comment]
@Krigpl 

Which logs suggest that? It has been created on today's morning and has not yet completed a single task.

**Comment by Krigpl** [comment]
- keys already exist (they are not generated)
- initial balance is not 0
- and the main culprit of this error - fundslocker restores task locks, meaning task lock existed before this run

**Comment by mplebanski** [comment]
@Krigpl 

Ok what shall we do about this task lock?

**Comment by Krigpl** [comment]
@mplebanski 
Just remove `fundslockerv2.pickle` file until the root issue is resolved.

**Comment by Krigpl** [comment]
No repro steps and probably no longer relevant.
